src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -155,6 +156,7 @@ private void checkOut(RoutingContext routingContext) {
       .thenApply(inactiveUserValidator::refuseWhenUserIsInactive)
       .thenApply(inactiveProxyUserValidator::refuseWhenUserIsInactive)
       .thenCombineAsync(itemRepository.fetchByBarcode(itemBarcode), this::addItem)
+      .thenComposeAsync(r -> r.after(this::addItemLocationIdAtCheckout))
```

Comment:
I think this statement has to be on line # 172. Because it can cause an NPE if item is not found by barcode.

Reply:
I believe it's on the right line now